src/main/java/com/rcv/ResultsWriter.java

```diff
@@ -643,48 +682,74 @@ void generateCdfJson(List<CastVoteRecord> castVoteRecords)
   private Map<String, Object> generateCvrSnapshotMap(
       CastVoteRecord cvr, Integer round, List<Pair<String, BigDecimal>> currentRoundSnapshotData) {
     List<Map<String, Object>> selectionMapList = new LinkedList<>();
-    for (int rank : cvr.rankToCandidateIDs.keySet()) {
-      for (String candidateCode : cvr.rankToCandidateIDs.get(rank)) {
-        String isAllocable = "unknown";
-        BigDecimal numberVotes = BigDecimal.ONE;
-        if (currentRoundSnapshotData != null) {
-          // scanning the list isn't actually expensive because it will almost always be very short
-          for (Pair<String, BigDecimal> allocation : currentRoundSnapshotData) {
-            if (allocation.getKey().equals(candidateCode)) {
-              isAllocable = "yes";
-              numberVotes = allocation.getValue();
-              break;
-            }
-          }
-          if (isAllocable.equals("unknown")) {
-            isAllocable = "no";
-            // not sure what numberVotes should be in this situation
+    List<Map.Entry<String, List<Integer>>> candidatesWithRanksList =
+        getCandidatesWithRanksList(cvr.rankToCandidateIDs);
+
+    for (Map.Entry<String, List<Integer>> candidateWithRanks : candidatesWithRanksList) {
+      String candidateCode = candidateWithRanks.getKey();
+
+      String isAllocable = "unknown";
+      BigDecimal numberVotes = BigDecimal.ONE;
+
+      if (currentRoundSnapshotData != null) {
+        // scanning the list isn't actually expensive because it will almost always be very short
+        for (Pair<String, BigDecimal> allocation : currentRoundSnapshotData) {
+          if (allocation.getKey().equals(candidateCode)) {
+            isAllocable = "yes";
+            numberVotes = allocation.getValue();
+            break;
           }
         }
-        Map<String, Object> selectionPositionMap =
-            Map.ofEntries(
-                entry("HasIndication", "yes"),
-                entry("IsAllocable", isAllocable),
-                entry("NumberVotes", numberVotes),
-                entry("Rank", rank),
-                entry("@type", "CVR.SelectionPosition"));
+        // didn't find an allocation, i.e. this ballot didn't contribute all or part of a vote to
+        // this candidate in this round
+        if (isAllocable.equals("unknown")) {
+          isAllocable = "no";
+          // not sure what numberVotes should be in this situation
```

Comment:
I'd just remove this

Reply:
Remove the comment, you mean? Or the field? The current behavior feels kind of weird.

Reply:
The current comment sounds like _maybe_ there is some remaining task here?  A bit confusing.
My take is, if it's not allocable, numberVotes is meaningless and it doesn't matter what happens to it.  I'd just remove it, or update it to explain why we have the current behavior.  It's a small thing, also feel free to leave as is.

Reply:
Your take is correct. As you said, it's meaningless. I had only included it because I thought it was required. I agree it would be better to remove it when isAllocable is false.

I'll leave it out of this PR because I want to get confirmation from John first... and once I make this change, I'll have to go back and update all the test output files yet again. :/

Follow-up in #266.